src/Command/SubProfileCommand.php

```diff
@@ -0,0 +1,363 @@
+      )
+      ->addOption(
+        'distribution',
+        false,
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('commands.generate.profile.options.distribution'),
+        false
+      );
+  }
+
+  /**
+   * ProfileCommand constructor.
+   */
+  public function __construct() {
+    $translater = new TranslatorManager();
+    $stringConverter = new StringConverter();
+    $renderer = new TwigRenderer($translater, $stringConverter);
+    $renderer->setSkeletonDirs([__DIR__ . '/../../templates/']);
+    $fileQueue = new FileQueue();
+    $generator = new SubProfileGenerator();
+    $generator->setRenderer($renderer);
+    $generator->setFileQueue($fileQueue);
+    $httpClient = new Client();
+    $appRoot = \Drupal::root();
+    $configurationManager = new ConfigurationManager();
+    $site = new Site($appRoot, $configurationManager);
+    $extensionManager = new Manager($site, $appRoot);
+    $validator = new Validator($extensionManager);
+    $this->excludedDependencies = [];
+    parent::__construct($extensionManager, $generator, $stringConverter, $validator, $appRoot, $site, $httpClient);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function interact(InputInterface $input, OutputInterface $output) {
+    $io = new DrupalStyle($input, $output);
+    $validators = $this->validator;
+
+    try {
+      // A profile is technically also a module, so we can use the same
+      // validator to check the name.
+      $profile = $input->getOption('profile') ? $validators->validateModuleName($input->getOption('profile')) : null;
+    } catch (\Exception $error) {
+      $io->error($error->getMessage());
+
+      return;
+    }
+
+    if (!$profile) {
+      $profile = $io->ask(
+        $this->trans('The name of the subprofile'),
+        '',
+        function ($profile) use ($validators) {
+          return $validators->validateModuleName($profile);
+        }
+      );
+      $input->setOption('profile', $profile);
+    }
+
+    try {
+      $machine_name = $input->getOption('machine-name') ? $validators->validateModuleName($input->getOption('machine-name')) : null;
+    } catch (\Exception $error) {
+      $io->error($error->getMessage());
+
+      return;
+    }
+
+    if (!$machine_name) {
+      $machine_name = $io->ask(
+        $this->trans('commands.generate.profile.questions.machine-name'),
+        $this->stringConverter->createMachineName($profile),
+        function ($machine_name) use ($validators) {
+          return $validators->validateMachineName($machine_name);
+        }
+      );
+      $input->setOption('machine-name', $machine_name);
+    }
+
+    $description = $input->getOption('description');
+    if (!$description) {
+      $description = $io->ask($this->trans('Subprofile description'), '');
+      $input->setOption('description', $description);
+    }
+
+    $dependencies = $input->getOption('dependencies');
+    if (!$dependencies) {
```

Comment:
We might want a more robust check here, since things like ' ' will be truthy. Do we need to run trim() on $dependencies or will Symfony do that for us?